src/main/java/com/puppycrawl/tools/checkstyle/checks/naming/GoogleMethodNameCheck.java

```diff
@@ -0,0 +1,194 @@
+     * char</li>
+     * <li>{@code (?![a-z][A-Z])} - Negative lookahead: cannot be like "fO"</li>
+     * <li>{@code [a-z]} - Must start with lowercase</li>
+     * <li>{@code [a-z0-9]*} - Followed by lowercase or digits</li>
+     * <li>{@code (?:[A-Z][a-z0-9]*)*} - CamelCase humps (uppercase followed by
+     * lowercase/digits)</li>
+     * <li>{@code $} - End of string (numbering suffix validated separately)</li>
+     * </ul>
+     */
+    private static final Pattern REGULAR_METHOD_NAME_PATTERN = Pattern
+            .compile("^(?![a-z]$)(?![a-z][A-Z])[a-z][a-z0-9]*(?:[A-Z][a-z0-9]*)*$");
+
+    /**
+     * Pattern for valid test method names in Google style.
+     * Each segment between underscores must:
+     * <ul>
+     * <li>Start with lowercase letter</li>
+     * <li>Be at least 2 characters long</li>
+     * <li>Not start with single lowercase followed by uppercase (e.g., "fO")</li>
+     * </ul>
+     */
+    private static final Pattern TEST_METHOD_NAME_PATTERN = Pattern.compile(
+            "^(?![a-z](?:_|$))(?![a-z][A-Z])[a-z][a-z0-9]*(?:[A-Z][a-z0-9]*)*"
+            + "(?:_(?![a-z](?:_|$))(?![a-z][A-Z])[a-z][a-z0-9]*(?:[A-Z][a-z0-9]*)*)*$");
+
+    /**
+     * Pattern to detect if name ends with a valid numbering suffix.
+     * A valid numbering suffix is underscore followed by digits only at the end.
+     */
+    private static final Pattern NUMBERING_SUFFIX_PATTERN = Pattern.compile("(?:_[0-9]+)+$");
+
+    /**
+     * Pattern to detect invalid underscore usage - underscore between letter and digit
+     * or between digit and letter.
+     */
+    private static final Pattern INVALID_UNDERSCORE_PATTERN =
+            Pattern.compile("[a-zA-Z]_[0-9]|[0-9]_[a-zA-Z]");
+
+    /**
+     * Set of JUnit 5 test annotation names that indicate a test method.
+     */
+    private static final Set<String> TEST_ANNOTATIONS = Set.of(
+            "Test",
+            "org.junit.jupiter.api.Test",
+            "org.junit.Test",
+            "ParameterizedTest",
+            "org.junit.jupiter.params.ParameterizedTest",
+            "RepeatedTest",
+            "org.junit.jupiter.api.RepeatedTest");
+
+    @Override
+    public int[] getDefaultTokens() {
+        return getRequiredTokens();
+    }
+
+    @Override
+    public int[] getAcceptableTokens() {
+        return getRequiredTokens();
+    }
+
+    @Override
+    public int[] getRequiredTokens() {
+        return new int[] {TokenTypes.METHOD_DEF};
+    }
+
+    @Override
+    public void visitToken(DetailAST ast) {
+        if (!AnnotationUtil.hasOverrideAnnotation(ast)) {
+            final DetailAST nameAst = ast.findFirstToken(TokenTypes.IDENT);
+            final String methodName = nameAst.getText();
+
+            if (hasTestAnnotation(ast)) {
+                validateTestMethodName(nameAst, methodName);
+            }
+            else {
+                validateRegularMethodName(nameAst, methodName);
+            }
```

Comment:
```suggestion
            if (!hasTestAnnotation(ast)) {
                validateRegularMethodName(nameAst, methodName);
            }
            else {
                validateTestMethodName(nameAst, methodName);
            }
```

More optimized because there will be a lot of regular methods compare to test methods.